src/app/components/dataset-properties-modal/dataset-properties-modal.component.ts

```diff
@@ -0,0 +1,212 @@
+  templateUrl: './dataset-properties-modal.component.html',
+  styleUrls: ['./dataset-properties-modal.component.scss']
+})
+export class DatasetPropertiesModal implements OnInit {
+
+  public datasetName = '';
+  public datasetCSIEntryType = '';
+  public datasetIsPDS = '';
+  public datasetMaxRecordLen = 0;
+  public datasetOrganization = '';
+  public datasetBlockSize = 0;
+  public datasetCarriageControl = '';
+  public datasetIsBlocked = '';
+  public datasetIsSpanned = '';
+  public datasetIsStandard = '';
+  public datasetRecordFormat = '';
+  public datasetVolser = '';
+  public datasetIcon = '';
+  public datasetSummary = '';
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) data,
+  ) 
+  {
+    const node = data.event;
+    if (node.data) {
+      const data = node.data;
+      this.datasetName = data.fileName;
+      this.datasetCSIEntryType = this.formatCSIEntryType(data.datasetAttrs.csiEntryType);
+      if (data.datasetAttrs.dsorg) {
+        if (data.datasetAttrs.dsorg.isPDSDir) {
+          if (data.datasetAttrs.dsorg.isPDSE) {
+            this.datasetIsPDS = "✓ (PDS/E)";
+          } else {
+            this.datasetIsPDS = "✓";
+          }
+        }
+        this.datasetOrganization = this.formatOrganization(data.datasetAttrs.dsorg.organization);
+        this.datasetBlockSize = data.datasetAttrs.dsorg.totalBlockSize;
+        this.datasetCarriageControl = data.datasetAttrs.recfm.carriageControl;
+        if (data.datasetAttrs.recfm.isBlocked) {
+          this.datasetIsBlocked = "✓";
+        } else {
+          this.datasetIsBlocked = "No";
+        }
+        if (data.datasetAttrs.recfm.isSpanned) {
+          this.datasetIsSpanned = "✓";
+        }
+        if (data.datasetAttrs.recfm.isStandard) {
+          this.datasetIsStandard = "✓";
+        }
+        this.datasetRecordFormat = this.formatRecordFormat(data.datasetAttrs.recfm.recordLength);
+        this.datasetMaxRecordLen = data.datasetAttrs.dsorg.maxRecordLen;
+        this.datasetVolser = data.datasetAttrs.volser;
+      }
+    }
+    if (node.icon) {
+      this.datasetIcon = node.icon;
+    } else if (node.collapsedIcon) {
+      this.datasetIcon = node.collapsedIcon;
+    }
+    this.datasetSummary = this.formatSummary(this.datasetOrganization, this.datasetRecordFormat, this.datasetMaxRecordLen);
+  }
+
+  ngOnInit() {
+  }
+
+  formatRecordFormat(recordFormat: string): string {
+    switch(recordFormat) {
+      case "U":
+        recordFormat = "U - Undefined"
+        break;
+      case "F":
+        recordFormat = "F - Fixed"
+        break;
+      case "V":
+        recordFormat = "V - Variable"
+        break;
+    }
+    return recordFormat;
+  }
+
+  formatSummary(org: string, recfm: string, reclen: number): string {
+    let summary = "N/A";
+    if (org.substring(0, 2) == "PS") {
+      if (recfm[0] == 'F') {
+        if (reclen == 80) {
+          summary = "FB80";
+        } else if (reclen == 256) {
+          summary = "FB256";
```

Comment:
Why not just do `FB+reclen`???
80 and 256 are common, but others are possible.

Reply:
Yes.